nixos/default.nix

```diff
@@ -9,35 +9,14 @@ let
     modules = [ configuration ];
   };
 
-  # This is for `nixos-rebuild build-vm'.
-  vmConfig = (import ./lib/eval-config.nix {
-    inherit system;
-    modules = [ configuration ./modules/virtualisation/qemu-vm.nix ];
-  }).config;
-
-  # This is for `nixos-rebuild build-vm-with-bootloader'.
-  vmWithBootLoaderConfig = (import ./lib/eval-config.nix {
-    inherit system;
-    modules =
-      [ configuration
-        ./modules/virtualisation/qemu-vm.nix
-        { virtualisation.useBootLoader = true; }
-        ({ config, ... }: {
-          virtualisation.useEFIBoot =
-            config.boot.loader.systemd-boot.enable ||
-            config.boot.loader.efi.canTouchEfiVariables;
-        })
-      ];
-  }).config;
-
 in
 
 {
   inherit (eval) pkgs config options;
 
   system = eval.config.system.build.toplevel;
 
-  vm = vmConfig.system.build.vm;
+  vm = eval.config.virtualisation.vmVariant.system.build.vm;
 
-  vmWithBootLoader = vmWithBootLoaderConfig.system.build.vm;
+  vmWithBootLoader = eval.config.virtualisation.vmVariantWithBootLoader.system.build.vm;
```

Comment:
Maybe we should deprecate those and use the `vmVariant`s directly in `nixos-rebuild`?

Reply:
Maybe at some point after a long transitioin period. They're canon: `nix-build <nixpkgs/nixos> -A vm` is a thing people do.

Reply:
Fair, that's a shorthand worth having.